### PR TITLE
Rename genesis block to genesis config

### PR DESCRIPTION
**Pull request by jstarry**
- Rename GenesisBlock and GenesisBlockInfo
- Change json rpc api from getGenesisBlockhash to getGenesisHash
- Change expected-genesis-blockhash arg to expected-genesis-hash

Part 1 of fix for https://github.com/solana-labs/solana/issues/6795

### CODE REVIEW

**Comment by garious** [comment]
The thing that you've renamed `GenesisConfigHash` looks to be a blockhash. Technically, a GenesisConfig hash could exist, but I don't think that's what's being returned.

File: book/src/api-reference/jsonrpc-api.md [hunk 1]
```diff
@@ -21,7 +21,7 @@ To interact with a Solana node inside a JavaScript application, use the [solana-
 * [getClusterNodes](jsonrpc-api.md#getclusternodes)
 * [getEpochInfo](jsonrpc-api.md#getepochinfo)
 * [getEpochSchedule](jsonrpc-api.md#getepochschedule)
-* [getGenesisBlockhash](jsonrpc-api.md#getgenesisblockhash)
+* [getGenesisConfigHash](jsonrpc-api.md#getgenesisconfighash)
```
Comment (mvines) [inline book/src/api-reference/jsonrpc-api.md]: I'd rather we keep this as ` getGenesisBlockhash`, and strike the downstream changes due to this rename.

Comment (jstarry) [reply book/src/api-reference/jsonrpc-api.md]: @mvines if we keep this as `getGenesisBlockhash` are we considering the genesis config's hash as a blockhash? Or is this being interpreted as the genesis block's hash? If it's the latter, I think we should return the hash of the first block here

**Comment by jstarry** [comment]
> The thing that you've renamed `GenesisConfigHash` looks to be a blockhash. Technically, a GenesisConfig hash could exist, but I don't think that's what's being returned.

@garious it looks like a "blockhash" from a bank's perspective but it is computed as a hash of the genesis config, there are no entries in it. I don't think it fits the definition here: https://github.com/solana-labs/solana/blob/master/book/src/terminology.md#blockhash

File: book/src/api-reference/jsonrpc-api.md (hunk 1 above)
Comment (mvines) [reply book/src/api-reference/jsonrpc-api.md]: genesis config's hash is a blockhash as in bank0 that's the only recent_blockhash available to transactions to use.

**Comment by mvines** [comment]
> > The thing that you've renamed `GenesisConfigHash` looks to be a blockhash. Technically, a GenesisConfig hash could exist, but I don't think that's what's being returned.
> 
> @garious it looks like a "blockhash" from a bank's perspective but it is computed as a hash of the genesis config, there are no entries in it. I don't think it fits the definition here: https://github.com/solana-labs/solana/blob/master/book/src/terminology.md#blockhash

Fix the definition?

**Comment by garious** [comment]
I'm lost, sorry. I don't see a problem with the blockhash definition.  Remind me why RPC needs to query for the genesis hash?

**Comment by garious** [comment]
Thinking more on it, calling the GenesisConfig hash a blockhash seems very wrong.

**Comment by jstarry** [comment]
> I'm lost, sorry. I don't see a problem with the blockhash definition. Remind me why RPC needs to query for the genesis hash?

We check it on validator boot up to make sure that a validator is starting up with the right genesis configuration

**Comment by jstarry** [comment]
won't fix - easier to update terminology

**Comment by garious** [comment]
Fixing the terminology comes with a fixed, known cost. The hardest part is coming to consensus, and that part is done. Not fixing terminology adds a cost to every engineer that encounters the inconsistency as well as every bug that arises around the confusion.

File: book/src/api-reference/jsonrpc-api.md (hunk 1 above)
Comment (jstarry) [reply book/src/api-reference/jsonrpc-api.md]: Changed to getGenesisHash as discussed in Discord

**Comment by solana-grimes** [comment]
:broken_heart: Unable to automerge due to CI failure

**Comment by solana-grimes** [comment]
:broken_heart: Unable to automerge due to CI failure

**Comment by solana-grimes** [comment]
:broken_heart: Unable to automerge due to CI failure

**Pull request merged**